src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,9 +45,13 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println();
-    RedditData post = new RedditData();
-    System.out.println(post.getRedditPosts());
+    try {
+      YoutubeApi.getService();
+      YoutubePost newPost = YoutubeApi.getYoutubePost();
+      response.setContentType("text/html;");
+      response.getWriter().println(newPost);
+    } catch (GeneralSecurityException | IOException e) {
+      System.out.println("Error: Youtube api returning exception" + e);
```

Comment:
Can you also write an error to the response and send that to the client side to handle?

Reply:
It was removed by mistake, added again!